pwa-devdocs/_drafts/state-management/index.md

```diff
@@ -0,0 +1,188 @@
+State management describes the way the application handles changing state data as a result of user interactions.
+
+Interactions, such as clicking on a button or loading the page, modify the state and update the appearance or behavior of the application.
+
+For example, a shopper clicks on a button to add an item to the shopping cart.
+The application needs a way to add that item to the shopping cart while the shopper continues to browse the application.
+It also needs to update the visual components that use shopping cart data to reflect the new state.
+
+## Local versus global state
+
+Local and global are the two different types of state a component can depend on.
+
+Local state data is any data scoped within a component or its children components.
+This type of data is not shared with a component's parent or peer data.
+Otherwise, that data should be [lifted][].
+
+For example, form data is never used outside of that components, so
+it is categorized as local state data.
+
+Global state data is any data made available to components in the entire application.
+Components that depend on a global state value subscribe to changes for that value and re-render themselves.
+Most components do not depend on the entire global state.
+Instead, a component only uses small pieces of the entire global state.
+
+Shopping cart data is an example of global state data that components in different levels of the application use and modify.
+
+## Common state management technologies
+
+There are many libraries and framework features that implement state management.
+This section describes two such technologies used in the PWA Studio project.
+
+### Redux
+
+[Redux][] is a state management design pattern and library.
+It promotes the idea of a global object tree that contains the state of the whole application.
+This object is known as a [store][].
+
+The store is a read-only object, which can only be updated by dispatching a [reducer][] function.
+Reducer functions accept the current state and an [action][] object as parameters and returns the next state.
+
+Application components are able to [dispatch][] various actions to update the state.
+Components can also [subscribe][] to state changes to update their appearance or behavior.
+
+Early versions of PWA Studio used the Redux library directly as the primary mechanism for managing application state,
+and the Redux pattern can be seen in hooks such as [`useQueryResult()`][] and [`useRestResponse()`][].
+
+Currently, PWA Studio abstracts away its Redux implementation details using Peregrine hooks and context providers.
+This opens up the possibility of the project replacing Redux in Peregrine with another state management library without breaking state dependent components, such as those in Venia.
+
+Since Peregrine components require a Redux store, you must create one in your storefront application project and provide reducers:
```

Comment:
I think a better way to explain this would be that we allow for customization of the Peregrine reducers/enhancers. In the below example you can see how you are allowed to provide your own reducers to the `combineReducers` function call. You may even want to include that in the example: `combineReducers(reducers, yourReducers)`